File: 34.在排序数组中查找元素的第一个和最后一个位置.cc
#include <algorithm>
#include <vector>
using std::find;
using std::lower_bound;
using std::upper_bound;
using std::vector;

class Solution {
public:
    vector<int> searchRange(vector<int> &nums, int target) {
        if (find(nums.begin(), nums.end(), target) == nums.end())
            return {-1, -1};
        int low = lower_bound(nums.begin(), nums.end(), target) - nums.begin();
        int up = upper_bound(nums.begin(), nums.end(), target) - nums.begin() - 1;
        return {low, up};
    }
};

File: Python/34.在排序数组中查找元素的第一个和最后一个位置.py
from typing import List
from bisect import bisect_left, bisect_right

class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        if target not in nums:
            return [-1, -1]
        left = bisect_left(nums, target)
        right = bisect_right(nums, target) - 1;
        return [left, right]
